docs/src/main/asciidoc/writing-extensions.adoc

```diff
@@ -809,6 +809,40 @@ SomeOtherBuildItem secondBuildStep() {
 }
 ----
 
+[id='validation-error-build-item']
+===== Validation Error build items
+
+They represent build items with validation errors that make the build fail.
+
+.Example of usage of an validation error build item in a "pseudo-target" style
+[source%nowrap,java]
+----
+@BuildStep
+void checkCompatibility(Capabilities capabilities, BuildProducer<ValidationErrorBuildItem> validationErrors) {
+    if (capabilities.isMissing(Capability.RESTEASY_REACTIVE)
+            && capabilities.isMissing(Capability.RESTEASY_CLASSIC)) {
+        validationErrors.produce(new ValidationErrorBuildItem(
+                new ConfigurationException("Can't use both Resteasy Classic and Reactive extensions at the same time!")));
```

Comment:
Very minor but as people are sometimes copy/pasting things and sticking to the style, I would rather make it:
```suggestion
                new ConfigurationException("Cannot use both RESTEasy Classic and Reactive extensions at the same time")));
```